core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/deletevolumes/DeleteVolumesService.java

```diff
@@ -0,0 +1,118 @@
+package com.sequenceiq.cloudbreak.core.flow2.cluster.deletevolumes;
+
+import static com.sequenceiq.cloudbreak.event.ResourceEvent.CLUSTER_DELETEDVOLUMES;
+import static com.sequenceiq.cloudbreak.event.ResourceEvent.CLUSTER_DELETINGVOLUMES;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.StackDeleteVolumesRequest;
+import com.sequenceiq.cloudbreak.cloud.CloudConnector;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudPlatformVariant;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.model.Platform;
+import com.sequenceiq.cloudbreak.cloud.model.Variant;
+import com.sequenceiq.cloudbreak.cluster.api.ClusterApi;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.CommonUtilService;
+import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
+import com.sequenceiq.cloudbreak.domain.Resource;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.reactor.api.event.resource.DeleteVolumesRequest;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.resource.ResourceService;
+import com.sequenceiq.cloudbreak.template.model.ServiceComponent;
+import com.sequenceiq.common.api.type.ResourceType;
+
+@Service
+public class DeleteVolumesService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeleteVolumesService.class);
+
+    @Inject
+    private CloudbreakFlowMessageService flowMessageService;
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private ResourceService resourceService;
+
+    @Inject
+    private ClusterApiConnectors clusterApiConnectors;
+
+    @Inject
+    private CommonUtilService commonUtilService;
+
+    public void deleteVolumes(StackDeleteVolumesRequest payload) {
+        flowMessageService.fireEventAndLog(payload.getStackId(),
+                Status.UPDATE_IN_PROGRESS.name(),
+                CLUSTER_DELETINGVOLUMES,
+                payload.getGroup(),
+                payload.getStackId().toString());
+    }
+
+    public void detachResources(List<CloudResource> cloudResourcesToBeDetached, String cloudPlatform, AuthenticatedContext ac) throws Exception {
+        LOGGER.debug("Detaching volumes {}", cloudResourcesToBeDetached);
+        CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(Platform.platform(cloudPlatform), Variant.variant(cloudPlatform));
```

Comment:
Why don't we use the platform variant of the related `Stack` object from one layer upper?

Reply:
platformVariant in StackDto is also a String - which still needs to be parsed to get Platform.platform and Variant.variant to get CloudPlatformVariant. the cloudPlatform is also set in the request in the handler using "String cloudPlatform = stack.getCloudPlatform();" so its essentially the same

Reply:
I don't think that the Platform and the PlatformVariant is the same, for example you can check the AWS related one:
https://github.com/hortonworks/cloudbreak/blob/be4d7cd534b9599738a0027d2676f5e5636a268f/cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsConstants.java#L30